### PR TITLE
tools/tpm2_quote: Fix -G option to -g

### PR DESCRIPTION
**Pull request by lestums**
Fixes : #1315

Work is already done on `tpm2_certify`

Some files are left to be checked for the same kind of fix (-G while it can be -g) :

> tools/tpm2_hash.c
> tools/tpm2_quote.c
> tools/tpm2_sign.c
> tools/tpm2_import.c
> tools/tpm2_getmanufec.c
> tools/tpm2_createek.c
> tools/tpm2_create.c
> tools/tpm2_createprimary.c
> tools/tpm2_listpersistent.c
> tools/tpm2_createak.c
> tools/tpm2_encryptdecrypt.c
> tools/tpm2_loadexternal.c
> tools/tpm2_verifysignature.c

Do i proceed on the same PR ?

Signed-off-by: sebastien le stum <lestums@gmail.com>

### CODE REVIEW

**Comment by williamcroberts** [comment]
> Do i proceed on the same PR ?

Yes you can just keep pushing patches to your branch the PR is generated from, and the PR will keep updating.

**Comment by lestums** [comment]
Checked all the tools using -G/-g combo : Unfixed ones are either both relying on -g / -G or the -G is linked to object algorithm.

File: tools/tpm2_quote.c [hunk 1]
```diff
@@ -322,10 +322,10 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
         { "message",              required_argument, NULL, 'm' },
         { "pcrs",                 required_argument, NULL, 'p' },
         { "format",               required_argument, NULL, 'f' },
-        { "sig-hash-algorithm",   required_argument, NULL, 'G' }
+        { "sig-hash-algorithm",   required_argument, NULL, 'g' }
```
Comment (williamcroberts) [inline tools/tpm2_quote.c]: can you switch this to halg as well.

**Comment by williamcroberts** [comment]
@lestums can you also add entries to the changelog as well in this PR.

**Comment by williamcroberts** [comment]
Finishing work tracked on #1398

**Pull request merged**